### PR TITLE
Bring Your Pet to Work Day (Veteran Quirk)

### PR DESCRIPTION
**Pull request by Thlumyn**
## About The Pull Request

It's 2 am and I'm coding so I expect a lot of bugs, please yell them at me so I can fix :3

Veteran players can now choose a quirk that spawns them with a pet carrier in their hand, containing a simplemob/basic animal.

They can choose their pet from the quirks menu. 

Some of the animals are TOO BIG to fit in the carriers and can't be put back in once taken out, but I really like the idea of some cowboy spaceman riding their horse into work, so I left in the big animals for the time being. 

## How This Contributes To The Skyrat Roleplay Experience

Who doesn't love pets? A lot of roleplay opportunities show up from pets, such as: hunting down and suing somebody because their pet duct spider killed your pet chinchilla (rest in peace to the chinchillas that died in testing this PR). There was a lot of discussion about more veteran perks in the discord and this seemed to be the most popular one. 

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

image
image
(And upon discovering that horses reasonably aggro on you if you try to shove them back into a carrier, here is me getting beat to death on the Interlink by a pony)
image

</details>

## Changelog

:cl:
add: Added pet owner quirk to spawn with a pet in a pet carrier.
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
how did you read my mind i literally thought of this 3 hours before this pr was made

**Comment by nikothedude** [comment]
also vetlocking things to vetlock them is bad form, the only things that are vetlocked are things with reason to be vetlocked (like we cant trust everyone with it)

**Comment by nikothedude** [comment]
oh and ALSO this could probably be upstreamed, i think tg would love it

**Comment by Thlumyn** [comment]
> also vetlocking things to vetlock them is bad form, the only things that are vetlocked are things with reason to be vetlocked (like we cant trust everyone with it)

Honestly, I really don't trust everyone with spawning in with a pet; but largely, this quirk was discussed a while ago in https://discord.com/channels/596783386295795713/1150923274839146586 as an idea for a veteran quirk

File: modular_skyrat/modules/pet_owner/pet_owner.dm [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"Crab" = /mob/living/basic/crab,
+	"Frog" = /mob/living/basic/frog,
+	"Space lizard" = /mob/living/basic/lizard/space,
+	"Tegu" = /mob/living/basic/lizard/tegu,
+	"Mothroach" = /mob/living/basic/mothroach,
+	"White mouse" = /mob/living/basic/mouse/white,
+	"Gray mouse" = /mob/living/basic/mouse/gray,
+	"Brown mouse" = /mob/living/basic/mouse/brown,
+	"Bat" = /mob/living/basic/bat,
+	"Kiwi" = /mob/living/basic/kiwi,
```
Comment (vinylspiders) [inline modular_skyrat/modules/pet_owner/pet_owner.dm]: It is not a big deal, but you could alphabetize this

File: modular_skyrat/modules/pet_owner/pet_owner.dm [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+/datum/quirk/item_quirk/pet_owner/add_unique(client/client_source)
+	var/desired_pet = client_source?.prefs.read_preference(/datum/preference/choiced/pet_owner) || "Random"
+
+	if(desired_pet != "Random")
+		pet_type = GLOB.possible_player_pet[desired_pet]
+
+	if(pet_type == NONE) // Pet not set, we're picking one for them.
+		pet_type = pick(flatten_list(GLOB.possible_player_pet))
+
+	var/obj/item/pet_carrier/carrier = new /obj/item/pet_carrier
```
Comment (vinylspiders) [inline modular_skyrat/modules/pet_owner/pet_owner.dm]:
```suggestion
	var/obj/item/pet_carrier/carrier = new /obj/item/pet_carrier(get_turf(quirk_holder))
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]
Looks good to me otherwise (codewise)

File: modular_skyrat/modules/pet_owner/pet_owner.dm (hunk 2 above)
Comment (Thlumyn) [reply modular_skyrat/modules/pet_owner/pet_owner.dm]: oops, I made some edits right before you sent this trying to figure how to do this thing right here, this is a much better way, let me edit it to fix that

**Comment by nikothedude** [comment]
> > also vetlocking things to vetlock them is bad form, the only things that are vetlocked are things with reason to be vetlocked (like we cant trust everyone with it)
> 
> Honestly, I really don't trust everyone with spawning in with a pet; but largely, this quirk was discussed a while ago in https://discord.com/channels/596783386295795713/1150923274839146586 as an idea for a veteran quirk

A. that link doesnt seem to work
B. why dont you trust them with this? its literally just a simplemob, how can this possibly be abused? if the concern is that certain mobs give buffs, you can lock those animals behind being vetlocked, but something like a corgi is harmless

**Comment by Thlumyn** [comment]
> > > also vetlocking things to vetlock them is bad form, the only things that are vetlocked are things with reason to be vetlocked (like we cant trust everyone with it)
> > 
> > 
> > Honestly, I really don't trust everyone with spawning in with a pet; but largely, this quirk was discussed a while ago in https://discord.com/channels/596783386295795713/1150923274839146586 as an idea for a veteran quirk
> 
> A. that link doesnt seem to work B. why dont you trust them with this? its literally just a simplemob, how can this possibly be abused? if the concern is that certain mobs give buffs, you can lock those animals behind being vetlocked, but something like a corgi is harmless

Ah, sorry, that link is from the veteran channel discussion about this. 

I don't so much expect it to be abused for mechanical advantage as much so as: pets provide more RP possibilities but also some more opportunity for things to go LRP and I don't want to see the Interlink flooded with 30 abandoned pets roundstart.

**Comment by Thlumyn** [comment]
Heck, somebody refactored parrots while this was up and now its failing, let me fix that

**Comment by Thlumyn** [comment]
That being said: personally I don't have very strong opinions either way on this being a veteran quirk or not, but because the idea for it came from the discussion there (and it sounded like Ignari wanted this as a veteran thing), that's what I'm putting it as for now.

**Comment by HoodMilk** [comment]
See how it goes as vet only for a little while, if it's not a problem then maybe it could be opened up for everyone.

**Review by vinylspiders — APPROVED** [review]

File: modular_skyrat/modules/pet_owner/pet_owner.dm [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	"Mouse (gray)" = /mob/living/basic/mouse/gray,
+	"Mouse (brown)" = /mob/living/basic/mouse/brown,
+	"Parrot" = /mob/living/basic/parrot,
+	"Penguin" = /mob/living/basic/pet/penguin/emperor/neuter,
+	"Pig" = /mob/living/basic/pig,
+	"Pug" = /mob/living/basic/pet/dog/pug,
+	"Rabbit" = /mob/living/basic/rabbit,
+	"Sloth" = /mob/living/basic/sloth,
+	"Snake" = /mob/living/basic/snake,
+	"Space carp" = /mob/living/basic/carp/pet,
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/pet_owner/pet_owner.dm]: I feel like the carp might be a bit too much, they can do a decent amount of damage around the place. Admittedly, I'm not home right now, so I can't tell what its stats are.

File: modular_skyrat/modules/pet_owner/pet_owner.dm [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	"Space cat" = /mob/living/simple_animal/pet/cat/space,
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/pet_owner/pet_owner.dm]: The space animals are kinda meant to be rare/unique animals, same goes with the void puppy, the cake cat and the bread cat, amd I feel like that kinda makes them lose a bit of what makes them unique/interesting, i.e. the fact that they're rare.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Comment by desertmote** [comment]
I notice you removed the parrot as part of the remove rare/dangerous animals commit, but I don't think parrots are either! Those should be cool to add, I think. 👍

**Comment by Thlumyn** [comment]
> I notice you removed the parrot as part of the remove rare/dangerous animals commit, but I don't think parrots are either! Those should be cool to add, I think. 👍

A parrot can beat a space carp in 1 v 1

Also, on further thought - the ability for anybody to get a parrot and put a headset on it with almost no effort, just... seems like it would get old pretty fast. But, heck, I do love parrots myself.

**Comment by YakumoChen** [comment]
Would be easy to just make a parrot child that has less HP and is not as annoying for the purposes of this PR

**Comment by SpringSkipper** [comment]
I hate vetlocking in general. The idea that we should have more vetlocked perks just to have more vetlocked perks is deeply flawed.

**Comment by burgerenergy** [comment]
No, it should be vetlocked if only to drip feed the amount of random ass pets added to the station. It's a good quirk, mind, but the server where 80 people can have pets and drop them about as a meme versus the one with 15~ is very different.

**Comment by CandleJaxx** [comment]
> No, it should be vetlocked if only to drip feed the amount of random ass pets added to the station. It's a good quirk, mind, but the server where 80 people can have pets and drop them about as a meme versus the one with 15~ is very different.

easy fix to this is making it a higher-point quirk, honestly; make it like six points or something stupid, and zero for vets

**Comment by Thlumyn** [comment]
> Would be easy to just make a parrot child that has less HP and is not as annoying for the purposes of this PR

That'd be real cool; some other animals like carp I think we could have a nerfed version that'd work well as pets. But I don't want to add too much content like new animal subtypes to this at the moment, maybe something to be added in the future if this works out without being too problematic though.

**Comment by burgerenergy** [comment]
> > No, it should be vetlocked if only to drip feed the amount of random ass pets added to the station. It's a good quirk, mind, but the server where 80 people can have pets and drop them about as a meme versus the one with 15~ is very different.
> 
> easy fix to this is making it a higher-point quirk, honestly; make it like six points or something stupid, and zero for vets

(We're) making motions moving away from quirk costs and opening that up, so that wouldn't really work.

**Comment by YakumoChen** [comment]
can you make sure all the pets are neutered? On top of having 20 animals show up a shift, I don't think we need to add dozens of puppies and kittens on top.

**Review by GoldenAlpharex — APPROVED** [review]
Seems good to me!

**Comment by GoldenAlpharex** [comment]
Should be good for a quick like 1-day test-merge, just to make sure there isn't any glaring issues with this.

**Comment by FieryMorty** [comment]
I don't think that this should be locked behind Veteran status.

Otherwise, I like the PR a lot, it will be nice.

**Comment by GoldenAlpharex** [comment]
It's been test-merged, there doesn't seem to have been any issues, let's get this merged!

**Pull request merged**